src/Http/Authentication.Core/src/AuthenticationService.cs

```diff
@@ -146,6 +154,14 @@ public virtual async Task SignInAsync(HttpContext context, string scheme, Claims
                 throw new ArgumentNullException(nameof(principal));
             }
 
+            if (Options.RequireAuthenticatedSignIn)
+            {
+                if (principal.Identity == null || !principal.Identity.IsAuthenticated)
```

Comment:
These error conditions are quite different and warrant different messages.

Reply:
Sure split them into their own